cmd/soroban-rpc/lib/Makefile

```diff
@@ -0,0 +1,14 @@
+
```

Comment:
It is needed to obtain the default target, it could also be a script

Reply:
In other words, what's the alternative? (we can't simply remove the Makefile)

Reply:
I've reworked the way in which it's built and moved everything to the top Makefile. PTAL.

The only thing I have not been able to work out is how to avoid compiling libpreflight twice and differently through the `build` (which builds libpreflight through `cargo build`) and `libpreflight target (which does `cd cmd/soroban-rpc/lib/preflight && cargo build --target $(CARGO_BUILD_TARGET) --release`).

I think it's much better overall though.